packages/flutter/lib/src/rendering/image.dart

```diff
@@ -28,22 +32,47 @@ class RenderImage extends RenderBox {
     Color color,
     BlendMode colorBlendMode,
     BoxFit fit,
-    FractionalOffset alignment,
+    FractionalOffsetGeometry alignment: FractionalOffset.center,
     ImageRepeat repeat: ImageRepeat.noRepeat,
-    Rect centerSlice
-  }) : _image = image,
-      _width = width,
-      _height = height,
-      _scale = scale,
-      _color = color,
-      _colorBlendMode = colorBlendMode,
-      _fit = fit,
-      _alignment = alignment,
-      _repeat = repeat,
-      _centerSlice = centerSlice {
+    Rect centerSlice,
+    bool matchTextDirection: false,
+    TextDirection textDirection,
+  }) : assert(scale != null),
+       assert(repeat != null),
+       assert(alignment != null),
+       assert(matchTextDirection != null),
+       _image = image,
+       _width = width,
+       _height = height,
+       _scale = scale,
+       _color = color,
+       _colorBlendMode = colorBlendMode,
+       _fit = fit,
+       _alignment = alignment,
+       _repeat = repeat,
+       _centerSlice = centerSlice,
+       _matchTextDirection = matchTextDirection,
+       _textDirection = textDirection {
     _updateColorFilter();
   }
 
+  bool _needsResolution = true;
```

Comment:
You might want to use the nullity of _resolvedAlignment for this bit.  That also stops code from using stale values because they'll be nulled out.

Reply:
Ah, yes. That's the solution I was looking for. Thanks.